lib/spack/llnl/util/symlink.py

```diff
@@ -100,7 +102,8 @@ def _win32_is_junction(path):
     """
     Determines if a path is a win32 junction
     """
-    if os.path.islink(path):
+    path = Path(path)
+    if path.is_symlink():
```

Comment:
I'm surprised the `GetFileAttributes(path)` call on line 119 succeeds since you're giving it a `Path` now and not a `str`. Does it indeed still work? If not, would this also work?

```suggestion
    if Path(path).is_symlink():
```

Reply:
That's a win32 api call so it most certainly does not work with pathlib, nice catch. The suggestion works.